core/rawdb/databases_64bit.go

```diff
@@ -29,8 +30,8 @@ const PebbleEnabled = true
 
 // NewPebbleDBDatabase creates a persistent key-value database without a freezer
 // moving immutable chain segments into cold storage.
-func NewPebbleDBDatabase(file string, cache int, handles int, namespace string, readonly bool, logger *log.Logger) (ethdb.Database, error) {
-	db, err := pebble.New(file, cache, handles, namespace, readonly, logger)
+func NewPebbleDBDatabase(file string, cache int, handles int, namespace string, readonly bool, logger *log.Logger, location common.Location) (ethdb.Database, error) {
```

Comment:
I like the addition of location to the db, but it feels unrelated to the definition of work objects. Any chance you can split into its own commit?